registry/infrastructure/repositories/base_repository.py

```diff
@@ -35,3 +42,17 @@ def add_all_items(self, items):
         except Exception as e:
             self.session.rollback()
             raise e
+
+    def test_db_connection(self, count=1):
+        try:
+            self.session.query("1").from_statement(text("SELECT 1")).all()
+            return True
+        except OperationalError as e:
+            if count == 1:
```

Comment:
Why are we using Count as an integer? I guess we should call it as retry with values as True or False.

Reply:
Also did we check if SqlAlchemy provides an option to test the connection? We should look into that

Reply:
See https://docs.sqlalchemy.org/en/13/core/engines.html#sqlalchemy.create_engine.params.pool_pre_ping

Reply:
Count needs to be an integer, as we will allow a limited no of retries.
```
if count <= MAX_RETRIES_ALLOWED:
  engine = create_engine(connection_string, echo=True)
  Session = sessionmaker(bind=engine)
  self.session = Session()
  return self.test_db_connection(count=count+1)
else:
  logger.info(f"Unable to connect to the database:: {repr(e)}")
  raise e
```

Reply:
With `pool_pre_ping=True`, it will test the connection whenever a new lambda container comes up. I won't test the connection lambda on every connection. In case of error, it will try to reconnect post raising the DB error. Then it will invalidate other connections in the database.